scheduler/src/cook/mesos/sandbox.clj

```diff
@@ -149,44 +156,37 @@
             (contains? #{:success :threshold} reason)
             (update :host->consecutive-failures dissoc hostname))))
 
-(defn- remove-task-ids-with-sandbox
-  "Filters the task-ids which have a sandbox directory from task-id->sandbox."
-  [datomic-conn task-id->sandbox]
-  (letfn [(retrieve-task-ids-with-sandbox [datomic-db task-ids]
-            (d/q '[:find [?t ...]
-                   :in $ [?t ...]
-                   :where
-                   [?e :instance/task-id ?t]
-                   [?e :instance/sandbox-directory ?s]]
-                 datomic-db task-ids))
-          (remove-task-id-with-sandbox [task-id->sandbox task-id]
-            (dissoc! task-id->sandbox task-id))
-          (update-filter-entries-histogram! [task-id->sandbox]
-            (histograms/update! sandbox-updater-filter-entries (count task-id->sandbox))
-            task-id->sandbox)]
-    (timers/time!
-      sandbox-updater-filter-duration
-      (->> (keys task-id->sandbox)
-           (retrieve-task-ids-with-sandbox (d/db datomic-conn))
-           (reduce remove-task-id-with-sandbox (transient task-id->sandbox))
-           persistent!
-           update-filter-entries-histogram!))))
+(defn- aggregate-unprocessed-task-ids!
+  "Aggregates the unprocessed task-id->sandbox-directory into the unprocessed-task-id->sandbox-atom."
+  [unprocessed-task-id->sandbox-atom task-id->sandbox-directory]
+  (swap! unprocessed-task-id->sandbox-atom
+         (fn [unprocessed-task-id->sandbox-in-atom]
+           (let [result-map (merge unprocessed-task-id->sandbox-in-atom task-id->sandbox-directory)]
+             (threadsafe-counter-reset sandbox-updater-unfiltered-count (count result-map))
+             result-map))))
+
+(defn- remove-processed-task-ids!
+  "Removes the processed-task-ids from the unprocessed-task-id->sandbox-atom."
+  [unprocessed-task-id->sandbox-atom processed-task-ids]
+  (swap! unprocessed-task-id->sandbox-atom
+         (fn [unprocessed-task-id->sandbox-in-atom]
+           (let [result-map (apply dissoc unprocessed-task-id->sandbox-in-atom processed-task-ids)]
+             (threadsafe-counter-reset sandbox-updater-unfiltered-count (count result-map))
+             result-map))))
 
 (defn refresh-agent-cache-entry
   "If the entry for the specified agent is not cached:
    - Triggers building an indexed version of all task-id to sandbox directory for the specified agent;
    - After the indexed version is built, the tasks which have sandbox directories are removed;
    - The filtered version of tasks without sandbox directories is then synced into the task-id->sandbox-agent.
    The function call is a no-op if the specified agent already exists in the cache."
```

Comment:
I think this docstring is now misleading, correct?

Reply:
Updated the docstring.